kuma/wiki/tests/test_utils.py

```diff
@@ -0,0 +1,290 @@
+import datetime
+import os.path
+
+from django.core.exceptions import ImproperlyConfigured
+
+from constance.test import override_config
+import mock
+from googleapiclient.http import HttpMockSequence
+from googleapiclient.errors import HttpError
+
+from kuma.core.tests import KumaTestCase
+from kuma.users.tests import UserTestCase
+
+from . import revision
+from ..utils import analytics_upageviews, analytics_upageviews_by_revisions
+
+
+GA_TEST_CREDS = r"""{
+  "type": "service_account",
+  "project_id": "test-suite-client",
+  "private_key_id": "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+  "private_key": "-----BEGIN PRIVATE KEY-----\n-----END PRIVATE KEY-----\n"
+}
+"""
+
+
+class RecordingHttpMockSequence(HttpMockSequence):
+    def __init__(self, iterable):
+        super(RecordingHttpMockSequence, self).__init__(iterable)
+        self.request_calls = []
+
+    def request(self, *args, **kwargs):
+        self.request_calls.append((args, kwargs))
+        return super(RecordingHttpMockSequence, self).request(*args, **kwargs)
+
+
+class AnalyticsUpageviewsTests(KumaTestCase):
+    @classmethod
+    def setUpClass(cls):
+        super(AnalyticsUpageviewsTests, cls).setUpClass()
+
+        dir_path = os.path.dirname(os.path.realpath(__file__))
+        with open(os.path.join(dir_path, 'analyticsreporting-discover.json')) as f:
+            cls.valid_discovery = f.read()
+
+        cls.start_date = datetime.date(2016, 1, 1)
+
+        cls.valid_response = """{"reports": [
```

Comment:
These are static values, and don't need to be in ``setUpClass``. They could just be declared as class-level variables.